### PR TITLE
dynamodb-lock-client: add retryCount to GenericConfig interface

### PR DESCRIPTION
**Pull request by mightypanders**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests). This fails with an Out of Memory exception although the tests were not touched and seem to have been submitted originally in this state.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [FailClosedConfig](https://github.com/tristanls/dynamodb-lock-client/commit/8c52936f2feecc33d09a64f4d94a8482f825ad42#diff-5db02ad11a9304a29393e6e12c22a0cddef236728974e52a21c66eabd1581022) and [FailOpenConfig](https://github.com/tristanls/dynamodb-lock-client/commit/8c52936f2feecc33d09a64f4d94a8482f825ad42#diff-f617b7bd938306a8215c9dd6fc47c15547791314010438a7d4a50e34ab418353) include retryCount
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/dynamodb-lock-client/index.d.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ interface GenericConfig {
     partitionKey: string;
     sortKey?: string | undefined;
     owner?: string | undefined;
+    retryCount?: number | undefined
```
Comment (RyoshiKayo) [inline types/dynamodb-lock-client/index.d.ts:22]: nit: missing semi-colon

File: types/dynamodb-lock-client/dynamodb-lock-client-tests.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import * as AWS from "aws-sdk";
-import * as DynamoDBLockClient from "dynamodb-lock-client";
+import  { DynamoDB } from 'aws-sdk'
```
Comment (RyoshiKayo) [inline types/dynamodb-lock-client/dynamodb-lock-client-tests.ts:1]: nit: missing semi-colon

**Review by giantcow — CHANGES_REQUESTED** [review]
I was able to run the tests locally, but there are requested changes:

```
# kayo @ DESKTOP in ~/workspace/DefinitelyTyped on git:master o [7:07:43] 
$ gh pr checkout 57386
remote: Enumerating objects: 15, done.
remote: Counting objects: 100% (9/9), done.
remote: Total 15 (delta 9), reused 9 (delta 9), pack-reused 6
Unpacking objects: 100% (15/15), 3.41 KiB | 139.00 KiB/s, done.
From https://github.com/DefinitelyTyped/DefinitelyTyped
 * [new ref]               refs/pull/57386/head -> mightypanders/master
Switched to branch 'mightypanders/master'

# kayo @ DESKTOPin ~/workspace/DefinitelyTyped on git:mightypanders/master o [7:08:48] 
$ npm test dynamodb-lock-client

> definitely-typed@0.0.3 test
> dtslint types "dynamodb-lock-client"

Installing to /home/kayo/.dts/typescript-installs/4.4...
Installed!

Installing to /home/kayo/.dts/typescript-installs/4.5...
Installed!

Installing to /home/kayo/.dts/typescript-installs/4.6...
Installed!

Error: /home/kayo/workspace/DefinitelyTyped/types/dynamodb-lock-client/dynamodb-lock-client-tests.ts:1:1
ERROR: 1:1    import-spacing  Too many spaces after 'import'
ERROR: 1:27   expect          TypeScript@4.6 compile error: 
Cannot find module 'aws-sdk' or its corresponding type declarations.
ERROR: 1:36   semicolon       Missing semicolon
ERROR: 13:16  whitespace      missing whitespace
ERROR: 38:16  whitespace      missing whitespace

/home/kayo/workspace/DefinitelyTyped/types/dynamodb-lock-client/index.d.ts:1:25
ERROR: 1:25   dt-header       Error parsing header. Expected: foo MAJOR.MINOR (patch version not allowed). See: https://github.com/Microsoft/dtslint/blob/master/docs/dt-header.md
ERROR: 8:26   expect          TypeScript@4.6 compile error: 
Cannot find module 'aws-sdk' or its corresponding type declarations.
ERROR: 22:36  semicolon       Missing semicolon

    at /home/kayo/workspace/DefinitelyTyped/node_modules/dtslint/bin/index.js:207:19
    at Generator.next (<anonymous>)
    at fulfilled (/home/kayo/workspace/DefinitelyTyped/node_modules/dtslint/bin/index.js:6:58)
```

**Comment by mightypanders** [comment]
> I was able to run the tests locally, but there are requested changes:

I did get them to run in the end, the issue was node which ran out of memory.  
I had to increase the 'old-space-size' to 3 GB (value determined by trial-and-error) like so
```
export NODE_OPTIONS="--max-old-space-size=3072"
```

With this variable set, the tests passed on my machine as well.

**Pull request closed without merging**